awx/main/tasks.py

```diff
@@ -1917,6 +1928,60 @@ def build_env(self, inventory_update, **kwargs):
         env['AWX_PRIVATE_DATA_DIR'] = kwargs.get('private_data_dir', '')
         return env
 
+    def get_inventory_plugin_name(self, inventory_update):
+        if inventory_update.source in Inventory.SOURCE_PLUGINS:
+            return Inventory.SOURCE_PLUGINS[inventory_update.source]
+        if inventory_update.source == 'custom':
+            # TODO: allow user to specify which plugin to use
+            return None  # intentionally allow other defaults, like ini
+        return 'script'
+
+    def build_inventory(self, inventory_update, **kwargs):
+        src = inventory_update.source
+        if src in Inventory.SOURCE_PLUGINS and Version(kwargs['ansible_version']) >= Version('2.4.0'):
+            cred_data = kwargs.get('private_data_files', {}).get('credentials', '')
+            cloud_credential = cred_data.get(inventory_update.credential, '')
+            # Build inventory contents specific to each plugin type
+            if src == 'openstack':
+                if (not inventory_update.credential) or inventory_update.credential.kind != 'openstack':
+                    raise RuntimeError('Openstack plugin needs corresponding openstack credential')
```

Comment:
I don't think we had this sort of validation before now - is this preferable to letting the plugin fail?